pkg/proxy/nftables/proxier.go

```diff
@@ -795,6 +809,42 @@ func (proxier *Proxier) OnNodeDelete(node *v1.Node) {
 func (proxier *Proxier) OnNodeSynced() {
 }
 
+func isIPv6CIDRString(cidr string) bool {
+	ip, _, _ := net.ParseCIDR(cidr)
+	return ip != nil && ip.To4() == nil
+}
+
+// OnServiceCIDRAdd is called whenever creation of new ServiceCIDR object
+// is observed.
+func (proxier *Proxier) OnServiceCIDRAdd(serviceCIDR *networking.ServiceCIDR) {
+	proxier.mu.Lock()
+	defer proxier.mu.Unlock()
+
+	for _, cidr := range serviceCIDR.Spec.CIDRs {
+		if proxier.ipFamily == v1.IPv6Protocol && isIPv6CIDRString(cidr) {
+			proxier.serviceCIDRs.Insert(cidr)
+		}
+		if proxier.ipFamily == v1.IPv4Protocol && !isIPv6CIDRString(cidr) {
+			proxier.serviceCIDRs.Insert(cidr)
+		}
+	}
+	if proxier.isInitialized() {
+		proxier.Sync()
+	}
+}
+
+// OnServiceCIDRUpdate is called whenever modification of an existing ServiceCIDR
+// object is observed.
+func (proxier *Proxier) OnServiceCIDRUpdate(_, _ *networking.ServiceCIDR) {}
```

Comment:
you need to handle these methods too

Reply:
yes, was prioritizing to get core logic (nft part) right first.